src/game/option.c

```diff
@@ -591,9 +550,7 @@ void DoDetailOptionHW(INVENTORY_ITEM *inv_item)
             max_row = DETAIL_HW_UI_BAR_SCALE;
         } else {
             static char tmp[10];
-            sprintf(
-                tmp, "%dx%d", AvailableResolutions[GameHiRes].width,
-                AvailableResolutions[GameHiRes].height);
+            sprintf(tmp, "%dx%d", GetScreenWidth(), GetScreenHeight());
```

Comment:
The in-game resolution changes and saves correctly, but it always shows "640x480" for me.

Reply:
Fixed